portfolio/src/main/webapp/script.js

```diff
@@ -6,10 +8,19 @@ function addRandomQuote() {
       ['“Art washes away from the soul the dust of everyday life.” --Pablo Picasso',
        '“Fiction, because it is not about somebody who actually lived in the real world, always has the possibility of being about oneself.” --Orson Scott Card',
        '“Somewhere, something incredible is waiting to be known.” --Carl Sagan',
-       '“Imagination is more important than knowledge. Knowledge is limited. Imagination encircles the world.” --Albert Einstein'];
+       '“Imagination is more important than knowledge. Knowledge is limited. Imagination encircles the world.” --Albert Einstein',
+       '“Any sufficiently advanced technology is indistinguishable from magic.” --Arthur C. Clarke'];
 
   // Pick a random quote.
-  const quote = quotes[Math.floor(Math.random() * quotes.length)];
+  var newQuoteNum = Math.floor(Math.random() * quotes.length);
+  
+  // Prevent the same quote from showing consecutively
+  if (newQuoteNum == quoteNum) {
+      ++newQuoteNum;
+      newQuoteNum %= quotes.length;
+  }
+  quoteNum = newQuoteNum;
```

Comment:
I would have this state transition as the last line of the function, so that it's more clear what the current state is for the rest of the function. (i.e. use newQuoteNum throughout the function and only update prevQuoteNum at the end.)

Reply:
Edited, thanks for the feedback.